.github/workflows/lint.yml

```diff
@@ -0,0 +1,34 @@
+name: Linting
+
+on: [pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+    - name: Npm install and build
+      # A "full" install is executed, since `npm ci` does not always exit
+      # with an error status code if the lock file is inaccurate.
+      #
+      # See: https://github.com/WordPress/gutenberg/issues/16157
+      run: |
+        npm install
+        npm run test:create-block
+        npm run build
+    - name: Lint JavaScript and Styles
+      run: npm run lint
+    - name: Lint ES5 built files (IE11)
+      run: npx eslint --parser-options=ecmaVersion:5 --no-eslintrc --no-ignore ./build/**/*.js
+    - name: Type checking
+      run: npm run build:package-types
+    - name: Build artifacts
+      run: npm run check-local-changes
+    - name: License compatibility
+      run: npm run check-licenses
```

Comment:
Those 2 could be still separate, they only need `npm ci`. On the other hand, they are quite fast. 🤔

Reply:
Yes, maybe too fast to deserve a separate job.